app/controllers/svg_sprite_controller.rb

```diff
@@ -12,10 +12,10 @@ def show
     no_cookies
 
     RailsMultisite::ConnectionManagement.with_hostname(params[:hostname]) do
-      theme_id = params[:theme_id].to_i
+      theme_id = params[:theme_id].to_i if params[:theme_id].present?
 
       if SvgSprite.version(theme_id) != params[:version]
-        return redirect_to path(SvgSprite.path(theme_id))
+        return redirect_to cdn_path(SvgSprite.path(theme_id))
```

Comment:
It looks like it has been `path` for a while; how come it has only recently broken?

Reply:
Was it because `theme_id` was converted to 0 if it was nil?

Reply:
Good question..... I think my change in https://github.com/discourse/discourse/commit/8e3691d5370bb95d99fe750f46287763721fcc9c changed something here. Let me investigate alittle more.